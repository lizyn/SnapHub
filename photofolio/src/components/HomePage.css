.user {
  /* background-color: cadetblue; */
  padding: 1em 0em 2em 0em;
  border-bottom: 1px solid slategrey;
}

.Avatar {
  float: left;
}

h3{
  float: right;
  width: 70%;
  margin-top: 2em;
  /* background-color: beige; */
}

.main {
  display: flex;
  /* background-color:olive; */
  margin: 8em 8em;
  margin-top: 5em;
  margin-bottom: 1em;
}

.users-section {
  display: flex;
  flex-direction: column;
  /* align-items: flex-start; */
  justify-content:flex-start;
  margin: 1em;
  flex-basis: 40%;
  /* background-color:antiquewhite; */
}

.right-section {
  display: flex;
  flex-direction: column;
  /* background-color: brown; */
  flex-basis: 60%;
  margin: 1em;
}

.right-top {
  display: flex;
  /* background-color:aquamarine; */
  height: 5em;
  justify-content: space-between;
}

.searchbar {
  /* background-color:cadetblue; */
  flex-basis: 21em;
  align-self: flex-end;
  order: 1;
  margin-left: 2em;
}

.searchbar input[type=text] {
  float:right;
  padding: 6px;
  margin-top: 8px;
  font-size: 14px;
  width: 16em;
  border: 1px solid slategrey;
  border-radius: 5px;
}

.searchbar button {
  float: left;
  padding: 4px 10px;
  margin: 6px 0px 0px 0px;
  background: rgb(255, 255, 255);
  font-size: 14px;
  border: 1px solid slategrey;
  cursor: pointer;
  border-radius: 10px;
  color:slategrey;
}

.searchbar img {
  width: 18px;
}

.newpost {
  /* background-color:burlywood; */
  flex-basis: 10em;
  align-self: flex-start;
  order: 2;
}

.feed {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  /* background-color: darksalmon; */
  margin-top: 1em;
}

.recommendations{
  display: flex;
  flex-direction: column;
  /* background-color:rosybrown; */
  margin-top: 2em;
  flex-basis: 40%;
}

.userList{
  margin-top: 1em;
  /* background-color: darkkhaki; */
  flex-basis: 20em;
  padding: 2em;
  border-radius: 10px;

}

p {
  font-family: 'Lato';
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 22px;
}

.newpost button {
  border: none;
  text-align:center;
  border-radius: 10px;
  font-size: medium;
}
.newPostSign {
  margin-right: 0.5em;
  /* height: 2.5em; */
  fill: rgb(255, 255, 255);
}
