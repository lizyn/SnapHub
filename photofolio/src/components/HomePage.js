import React from "react";
import Navbar from "./Navbar";
import './HomePage.css'
import { Avatar } from "@mui/material";
import userMe from "../images/userMe.jpg";
import searchIcon from '../icons/Search.svg'
import NewIcon from '../icons/New.svg'
import Fab from "@mui/material/Fab";
import { ThemeProvider, createTheme } from '@mui/material/styles'
import FeedList from "./FeedList";
import CreatePostModal from "./CreatePostModal";
import FollowRecommendation from "./FollowRecommendation";


function HomePage(props) {

  const orange = createTheme({
    status: {
      danger: '#e53e3e',
    },
    palette: {
      primary: {
        main: '#FFDD2B',
        darker: '#000',
      },
      neutral: {
        main: '#FFDD2B',
        contrastText: '#FFF',
      },
    },
  });

  return (
    <div className="flex1">
      <Navbar {...props} />
      <CreatePostModal closeModal={props.closePostModal} open={props.postModalIsOpen} />
      <div className="main">
        <div className="users-section">
          <div className="user">
            <Avatar alt='me' className="Avatar" src={userMe} sx={{ width: 100, height: 100 }}></Avatar>
            <h3>Tatiana Dokidis</h3>
          </div>
          <div className="recommendations">
            <p>Recommended for you</p>
            <div className="userList">
              <p>Recommneded Users to follow</p>
              <FollowRecommendation />
            </div>
          </div>
        </div>

        <div className="right-section">
          <div className="right-top">
            <div className="searchbar">
              <input type="text" placeholder="Tag, user, etc..." name="search"></input>
              <button type="submit"><img src={searchIcon} alt="search"></img> Search </button>
            </div>
            <div className="newpost" onClick={() => props.setPostModalOpen(o => !o)}>
              <ThemeProvider theme={orange}>
                <Fab variant="extended" color="primary" sx={{ mb: 20 }}><img src={NewIcon} className="newPostSign" alt="new"></img>New Post</Fab>
              </ThemeProvider>
            </div>
          </div>

          <div className="feed">
            <FeedList></FeedList>
          </div>
        </div>
      </div>
    </div>
  )
}

export default HomePage;