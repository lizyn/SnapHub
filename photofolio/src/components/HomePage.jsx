import React from 'react';
import PropTypes from 'prop-types';
import './HomePage.css';
import { Avatar } from '@mui/material';
import { Link } from 'react-router-dom';
import Fab from '@mui/material/Fab';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import FeedList from './FeedList';
import CreatePostModal from './CreatePostModal';
import searchIcon from '../icons/Search.svg';
import NewIcon from '../icons/New.svg';
import UserList from './UserList';

function HomePage(props) {
  HomePage.propTypes = {
    closePostModal: PropTypes.func.isRequired,
    postModalIsOpen: PropTypes.bool.isRequired,
    setPostModalOpen: PropTypes.func.isRequired,
    setAlert: PropTypes.func.isRequired
  };
  const { closePostModal, postModalIsOpen, setPostModalOpen, setAlert } = props;
  const orange = createTheme({
    status: {
      danger: '#e53e3e'
    },
    palette: {
      primary: {
        main: '#FFDD2B',
        darker: '#000'
      },
      neutral: {
        main: '#FFDD2B',
        contrastText: '#FFF'
      }
    }
  });

  return (
    <div className="flex1">
      <CreatePostModal
        closeModal={closePostModal}
        open={postModalIsOpen}
        setAlert={setAlert}
      />
      <div className="main">
        <div className="users-section">
          <div className="user">
            <Link to="/profile">
              <Avatar
                alt="me"
                className="Avatar"
                src="https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1185.jpg"
                sx={{ width: 100, height: 100 }}
              />
            </Link>
            <h3>Tatiana Dokidis</h3>
          </div>
          <div className="recommendations">
            <div style={{ marginBottom: '.5em' }}>Recommended for you</div>
            <UserList />
          </div>
        </div>

        <div className="right-section">
          <div className="right-top">
            <div className="searchbar">
              <input
                type="text"
                placeholder="Tag, user, etc..."
                name="search"
              />
              <button type="submit">
                <img src={searchIcon} alt="search" /> Search{' '}
              </button>
            </div>
            <div className="newpost">
              <ThemeProvider theme={orange}>
                <Fab
                  variant="extended"
                  color="primary"
                  sx={{ mb: 20, zIndex: 999 }}
                  onClick={() => setPostModalOpen((o) => !o)}
                >
                  <img src={NewIcon} className="newPostSign" alt="new" />
                  New Post
                </Fab>
              </ThemeProvider>
            </div>
          </div>

          <div className="feed">
            <FeedList />
          </div>
        </div>
      </div>
    </div>
  );
}

export default HomePage;
