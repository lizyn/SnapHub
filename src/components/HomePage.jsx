import { React, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import './HomePage.css';
import { Avatar } from '@mui/material';
import { Link } from 'react-router-dom';
import Fab from '@mui/material/Fab';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import FeedList from './FeedList';
import CreatePostModal from './CreatePostModal';
// import searchIcon from '../icons/Search.svg';
import NewIcon from '../icons/New.svg';
import UserList from './UserList';
import { fetchCurUser } from '../api/axios';

function HomePage(props) {
  HomePage.propTypes = {
    closePostModal: PropTypes.func.isRequired,
    postModalIsOpen: PropTypes.bool.isRequired,
    setPostModalOpen: PropTypes.func.isRequired,
    setAlert: PropTypes.func.isRequired
  };
  const { closePostModal, postModalIsOpen, setPostModalOpen, setAlert } = props;
  const orange = createTheme({
    status: {
      danger: '#e53e3e'
    },
    palette: {
      primary: {
        main: '#FFDD2B',
        darker: '#000'
      },
      neutral: {
        main: '#FFDD2B',
        contrastText: '#FFF'
      }
    }
  });
  const [user, setUser] = useState(null);

  useEffect(() => {
    async function getCurUser() {
      const token = sessionStorage.getItem('user');
      let userData = {};
      if (token) userData = await fetchCurUser(token);
      setUser(userData[0]);
    }
    getCurUser();
    if (!sessionStorage.getItem('app-token')) window.location.replace('/login');
  }, []);
  // console.log(sessionStorage.getItem('app-token'));

  if (
    !sessionStorage.getItem('user') ||
    !user ||
    !sessionStorage.getItem('app-token')
  )
    return <div />;

  return (
    <div className="flex1">
      <CreatePostModal
        closeModal={closePostModal}
        open={postModalIsOpen}
        setAlert={setAlert}
        curUserFirstName={`${user.firstName} ${user.lastName}`}
        curUserAvatar={user.avatar}
        // eslint-disable-next-line no-underscore-dangle
        curUserId={user._id}
      />
      <div className="main">
        <div className="users-section">
          <div className="user">
            <Link to="/profile">
              <Avatar
                alt="me"
                className="Avatar"
                src={user.avatar}
                sx={{ width: 100, height: 100 }}
              />
            </Link>
            <h3>
              {user.firstName && user.lastName
                ? `${user.firstName} ${user.lastName}`
                : user.username || 'User 42'}
            </h3>
          </div>
          <div className="recommendations">
            <div style={{ marginBottom: '.5em' }}>Recommended for you</div>
            <div
              style={{
                height: '40vh',
                overflow: 'auto',
                display: 'flex',
                flexDirection: 'column'
                // border: '1px solid silver',
                // borderRadius: '3px'
              }}
            >
              <UserList />
            </div>
          </div>
        </div>

        <div className="right-section">
          <div className="right-top">
            {/* <div className="searchbar">
              <input
                type="text"
                placeholder="Tag, user, etc..."
                name="search"
              />
              <button type="submit">
                <img src={searchIcon} alt="search" /> Search{' '}
              </button>
            </div> */}
            <div
              className="newpost"
              style={{
                margin: 'auto'
              }}
            >
              <ThemeProvider theme={orange}>
                <Fab
                  id="newposttest"
                  variant="extended"
                  color="primary"
                  sx={{ zIndex: 998 }}
                  onClick={() => setPostModalOpen((o) => !o)}
                >
                  <img src={NewIcon} className="newPostSign" alt="new" />
                  New Post
                </Fab>
              </ThemeProvider>
            </div>
          </div>

          <div className="feed">
            {/* eslint-disable-next-line no-underscore-dangle */}
            <FeedList
              // eslint-disable-next-line no-underscore-dangle
              curUserId={user._id}
              curUserName={`${user.firstName} ${user.lastName}`}
              curUserAvatar={user.avatar}
            />
          </div>
        </div>
      </div>
    </div>
  );
}

export default HomePage;
