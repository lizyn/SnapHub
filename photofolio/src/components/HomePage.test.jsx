/**
 * @jest-environment jsdom
 */
import { React } from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import HomePage from './HomePage';

test('HomePage component matches snapshot', () => {
  const component = renderer.create(
    <HomePage
      postModalIsOpen={false}
      setPostModalOpen={() => {}}
      closePostModal={() => {}}
      setAlert={() => {}}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CreateModal opens when create post fab clicked', async () => {
  render(
    <HomePage
      postModalIsOpen={false}
      setPostModalOpen={() => {}}
      closePostModal={() => {}}
      setAlert={() => {}}
    />
  );
  const createBtn = screen.getByRole('button', { name: 'new New Post' });
  await userEvent.click(createBtn);
  expect(screen.getByText(/New/)).toBeVisible();
});
