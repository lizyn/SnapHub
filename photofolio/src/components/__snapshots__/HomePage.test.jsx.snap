// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component matches snapshot 1`] = `
<div
  className="flex1"
>
  <div
    className="main"
  >
    <div
      className="users-section"
    >
      <div
        className="user"
      >
        <a
          href="/profile"
          onClick={[Function]}
        >
          <div
            className="MuiAvatar-root MuiAvatar-circular Avatar css-124yhl1-MuiAvatar-root"
          >
            <img
              alt="me"
              className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/1185.jpg"
            />
          </div>
        </a>
        <h3>
          Tatiana Dokidis
        </h3>
      </div>
      <div
        className="recommendations"
      >
        <div
          style={
            Object {
              "marginBottom": ".5em",
            }
          }
        >
          Recommended for you
        </div>
      </div>
    </div>
    <div
      className="right-section"
    >
      <div
        className="right-top"
      >
        <div
          className="searchbar"
        >
          <input
            name="search"
            placeholder="Tag, user, etc..."
            type="text"
          />
          <button
            type="submit"
          >
            <img
              alt="search"
              src={Object {}}
            />
             Search
             
          </button>
        </div>
        <div
          className="newpost"
        >
          <button
            className="MuiButtonBase-root MuiFab-root MuiFab-extended MuiFab-sizeLarge MuiFab-primary MuiFab-root MuiFab-extended MuiFab-sizeLarge MuiFab-primary css-1j8wu97-MuiButtonBase-root-MuiFab-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <img
              alt="new"
              className="newPostSign"
              src={Object {}}
            />
            New Post
          </button>
        </div>
      </div>
      <div
        className="feed"
      >
        <div
          className="feedlist-main"
        />
      </div>
    </div>
  </div>
</div>
`;
